pkg/projectfile/projectfile.go

```diff
@@ -464,6 +465,21 @@ func (p *Project) Init() error {
 	}
 	p.parsedURL = parsedURL
 
+	if p.parsedURL.LegacyCommitID != "" {
+		// Migrate from commitID in activestate.yaml to .activestate/commit file.
+		// Writing to disk during Parse() feels wrong though.
+		if err := localcommit.Set(filepath.Dir(p.Path()), p.parsedURL.LegacyCommitID); err != nil {
+			return errs.Wrap(err, "Could not create local commit file")
+		}
+		pf := NewProjectField()
+		if err := pf.LoadProject(p.Project); err != nil {
+			return errs.Wrap(err, "Could not load activestate.yaml")
+		}
+		if err := pf.Save(p.path); err != nil {
+			return errs.Wrap(err, "Could not save activestate.yaml")
+		}
```

Comment:
This is initializing a new Project struct and saving that. This should instead operate on the one we're on.

Reply:
Perhaps I'm missing something, but this is how it's done when setting namespace (https://github.com/ActiveState/cli/blob/4f1d3400cdffcf6b7feeee5ad4552129fb0ff40d/pkg/projectfile/projectfile.go#L702-L709) and branch (https://github.com/ActiveState/cli/blob/4f1d3400cdffcf6b7feeee5ad4552129fb0ff40d/pkg/projectfile/projectfile.go#L722-L735). The projectfile's project field does not appear to persist anywhere.

Reply:
Oh my bad, I was reading `pf` as being `Project` but this is that special struct we made that ONLY looks at the project field. Ignore my comment above.